src/components/kubectl/logs.ts

```diff
@@ -0,0 +1,184 @@
+    explorerNode: ResourceNode,
+    displayMode: LogsDisplayMode
+) {
+    const namespace = explorerNode.namespace;
+    const podSummary = { name: explorerNode.id, namespace: namespace };
+    const container = await selectContainerForPod(podSummary);
+
+    // No container. How'd we get here. How does this Pod even work. What is going on.
+    if (!container) {
+        return;
+    }
+
+    getLogsForContainer(
+        kubectl,
+        podSummary,
+        container.name,
+        displayMode
+    );
+
+    return;
+}
+
+/**
+ * Fetches logs for a pod. If there are more than one containers,
+ * prompts the user for which container to fetch logs for.
+ * @param kubectl
+ * @param podSummary
+ * @param displayMode
+ */
+async function getLogsForPod(kubectl: Kubectl, podSummary: PodSummary, displayMode: LogsDisplayMode) {
+    if (!podSummary) {
+        vscode.window.showErrorMessage('Can\'t find a pod!');
+        return;
+    }
+
+    const container = await selectContainerForPod(podSummary);
+    if (!container) {
+        return;
+    }
+
+    getLogsForContainer(kubectl, podSummary, container.name, displayMode);
+}
+
+/**
+ * Gets the logs for a container in a provided pod, in a provided namespace, in a provided container.
+ * @param kubectl
+ * @param podSummary
+ * @param containerName
+ * @param displayMode
+ */
+function getLogsForContainer(
+    kubectl: Kubectl,
+    podSummary: PodSummary,
+    containerName: string | undefined,
+    displayMode: LogsDisplayMode
+) {
+    let cmd = `logs ${podSummary.name}`;
+
+    if (podSummary.namespace) {
+        cmd = `${cmd} --namespace=${podSummary.namespace}`;
+    }
+
+    if (containerName) {
+        cmd = `${cmd} --container=${containerName}`;
+    }
+
+    if (displayMode === LogsDisplayMode.Follow) {
+        cmd = `${cmd} -f`;
+        kubectl.invokeInNewTerminal(cmd, FOLLOW_LOGS_TERMINAL);
+        return;
+    }
+
+    kubectl.invokeInSharedTerminal(cmd);
+}
+
+/**
+ * Searches for a pod yaml spec from the open document
+ * or from the currently selected namespace.
+ * @param kubectl
+ * @param displayMode
+ */
+async function findPod(kubectl: Kubectl, displayMode: LogsDisplayMode) {
+    const editor = vscode.window.activeTextEditor;
+
+    if (editor) {
+        return findPodFromOpenDocument(kubectl, editor, displayMode);
+    }
+
+    return await findPodFromCurrentNamespace(kubectl, displayMode);
+}
+
+/**
+ * Finds a Pod from the open editor.
+ * @param editor
+ * @param editor
+ * @param displayMode
+ */
+async function findPodFromOpenDocument(kubectl: Kubectl, editor: vscode.TextEditor, displayMode: LogsDisplayMode) {
+    const text = editor.document.getText();
+    try {
+        const obj: {} = yaml.safeLoad(text);
+        if (!isPod(obj)) {
+            return;
+        }
+
+        const podSummary = {
+            name: obj.metadata.name,
+            namespace: obj.metadata.namespace
+        };
+
+        getLogsForPod(kubectl, podSummary, displayMode);
```

Comment:
Hmm, seems like we have a naming issue here - `findPodFromOpenDocument` implies it finds a pod, it says nothing about fetching logs.  Of course the naming issue is very likely inherited from `extension.ts` but it might be good to tidy this if we are doing a refactor.

Reply:
👍